tools/codegen/gen_functionalization_type.py

```diff
@@ -129,11 +136,11 @@ def emit_view_functionalization_body(
     assert_view_op_properties(f.func)
     view_tensor_name = dispatcher_sig.arguments()[0].name
 
-    keyset = 'dispatchKeySet & c10::after_func_keyset'
     return_type = dispatcher_sig.returns_type().remove_const_ref().cpp_type()
 
-    unwrap_tensor_args_str, unwrapped_args_ctx = unwrap_tensor_args(dispatcher_sig)
-    view_redispatch_args = [keyset] + [e.expr for e in translate(unwrapped_args_ctx, call_sig.arguments(), method=False)]
+    # For view ops, we're not using the data from the input tensors so we don't need to sync inputs.
```

Comment:
oh, commented here

Reply:
It took me a few read throughs to understand this comment. What you appear to be saying is that for views, we don't actually do any compute/copy; we just create a new view object that still points back to the base. So we can continue to defer syncing, since an access to the view object itself will cause a sync. (Importantly, no translation from view to view_copy occurs yet).

Reply:
yep, phrased a bit poorly :). I'll make it clearer

Reply:
Although technically I don't think we'll need to sync when we convert to `{view}_copy` operators either - we could just copy the stale data, and also copy any of the pending updates from the input `FunctionalTensorWrapper` onto the output. That way when we eventually sync the (stale) output at a later point, all of the previous updates are included too (this is what `pytorch/xla` does today - the optimization where if you create a view and never use the base again, then any mutations on the view don't actually need to get run on the base).